File: ShallowWaterSolvers/compressible/rossbywave/cacheSphi1.cpp
#include "nr.h"
#include "parameters.h"

void NR::cacheSphi1(Mat_O_DP &Sphi1, Vec_I_DP &phi)
{
	// Computes the values of the cosine phi1 basis
	// functions at each collocation point.

	for (int n=1;n<=N+1;n++)
	{
		for (int i=1;i<=N;i++)
		{
			Sphi1[n-1][i-1]=sin((2.0*n-1.0)*phi[i-1]);
		}
	}
}

File: MatlabCodeAndResults/incompressible nonliear shallow water equations/Fourier Series methods/no eta in ulam zonal flow/cacheSphi1.m
function Sphi1 = cacheSphi1(N,phi)

%// Computes the values of the cosine basis
%// functions at each collocation point.

Sphi1=zeros(N+1,N);

for n=1:N+1
	for i=1:N
		Sphi1(n,i)=sin((2*n-1)*phi(i,1));
    end
end